internal/dataplane/sendconfig/inmemory.go

```diff
@@ -66,6 +66,11 @@ func (s UpdateStrategyInMemory) Update(ctx context.Context, targetState ContentW
 
 	flattenErrors := shouldUseFlattenedErrors(s.version)
 	if errBody, err := s.configService.ReloadDeclarativeRawConfig(ctx, bytes.NewReader(config), true, flattenErrors); err != nil {
+		if len(errBody) == 0 {
+			// If we failed to get response in reloading config, we put the error from reloading in the parseErr return value
+			// to generate a proper error log.
+			return err, nil, err
+		}
 		resourceErrors, parseErr := parseFlatEntityErrors(errBody, s.log)
```

Comment:
Wouldn't it be better to handle this in parseFlatEntityErrors?

Reply:
The only place where `parseError` is here:
```golang
	if parseErr != nil {
		log.WithError(parseErr).Error("failed parsing resource errors")
		return nil
	}
```
How about returning `nil` here and omit the log when no response get? This might emit better log than returning a fake "parse error".

Reply:
moved to `parseFlatEntityErrors`.